### ISSUE TITLE
mouse not sharing after upgrading from Ubuntu 20.04 to Ubuntu 22.04

### ISSUE THREAD

**Issue by hamkuu** [debauchee/barrier#1910]
### What happened?

I have two computers both running Ubuntu 20.04,
one as the server, the other as the client, and Barrier v2.4 worked fine for me over a year.

After upgrading both computer to Ubuntu 22.04, and installing Barrier v2.4,
and the client says "connected to server".

But the mouse on the server cannot move to the client.

### Version

v2.4.0

### If applicable, where did you install Barrier from?

downloaded from Ubuntu Software.

### What OSes are you seeing the problem on? (Check all that apply)

Linux

### What OS versions are you using?

Ubuntu 22.04 LTS

### Relevant log output

```shell
[2023-03-03T10:29:33] INFO: starting server
[2023-03-03T10:29:33] INFO: config file: /tmp/Barrier.ab9639
[2023-03-03T10:29:33] INFO: log level: INFO
started server (IPv4/IPv6), waiting for clients
[2023-03-03T10:29:33] NOTE: accepted client connection
[2023-03-03T10:29:33] NOTE: client "MyClientHostname" has connected
```

**Comment by SlowHyper** [comment]
Upgraded Ubuntu from 20.04 to 22.04 and barrier server was not working. I disabled SSL but still barrier was not working.
For anyone interested my problem was solved and SSL is now working fine, by logging in with an xorg session.
Maybe wayland support is the problem! I hope this helps.

**Comment by eliumoraes** [comment]
I followed SlowHyper instructions and it worked for me! Thanks for sharing!

**Comment by alanjames1987** [comment]
I'm getting the same error. The computers connect but cannot control each other.

I'm running Kubuntu  22.04 on my desktop / server and Ubuntu 22.04 on my laptop / client. I've installed Wayland on my desktop using `sudo apt-get install plasma-workspace-wayland` and switched to it by logging out, selecting the Wayland option in the bottom left, and logging back in. 

When running X11 or Wayland on my desktop I can move the mouse off screen but the client mouse will not move. The keyboard is controllable however.

I've also reversed this, running the server on the laptop running Wayland and the client on the desktop running X11. In this case when moving the mouse off screen of the server I can't control anything on the client.

**Comment by eliumoraes** [comment]
> ```shell
> [2023-03-03T10:29:33] INFO: starting server
> [2023-03-03T10:29:33] INFO: config file: /tmp/Barrier.ab9639
> [2023-03-03T10:29:33] INFO: log level: INFO
> started server (IPv4/IPv6), waiting for clients
> [2023-03-03T10:29:33] NOTE: accepted client connection
> [2023-03-03T10:29:33] NOTE: client "MyClientHostname" has connected
> ```

Did you open the ports on both sides (client and server) and checked if the machine names are the same in the configurations? It's also important for you to verify if the SSH key is properly configured.

**Comment by eliumoraes** [comment]
I also usually set a fixed IP address for the server because it makes it easier for clients to locate the machine.

**Comment by andydude** [comment]
Same issue, my server says "client ... is connected" and my client says "connected to server", but the mouse doesn't travel between monitors like it used to. I'm not sure what "logging into Xorg" means, but I suspect it means that I'm running Xwayland. I have no idea how to configure that, and would prefer to not have to reinstall Ubuntu with a different window manager just to use one app. Does barrier support wayland?

**Comment by sgon00** [comment]
>  Does barrier support wayland?

@andydude No, barrier does NOT support wayland. I think the project is dead.

**Comment by jamie-pate** [comment]
for me it works on the right side of the screen but not the left, but even though the mouse travels, it doesn't really work
